aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1899,20 +1929,17 @@ TORCH_IMPL_FUNC(bmm_out_cpu)
     }
 }
 
-Tensor& dot_out(const Tensor& self, const Tensor& other, Tensor& result) {
-  auto output_device = result.device();
-  auto input1_device = self.device();
-  auto input2_device = other.device();
-  // check if the input & output tensors are on the same device.
-  TORCH_CHECK(
-    (output_device == input1_device) && (input1_device == input2_device),
-    "dot: Expected the output and input tensors to be on the "
-    "same device, but got the output tensor on ", output_device,
```

Comment:
Seems like this device check is lost? Or is it captured somewhere else?

Reply:
from what i've seen from the generated code in `build/aten/src/ATen/Register*.cpp`, making an op structured doesn't add any automatic device checks, it'll just add the scaffolding to be able to write a `TORCH_META_FUNC` & do what you want in it (which may include a device check).

the automatic device check generation seems to be controlled by whether you have `device_check: NoCheck` in the entry in native_functions.yaml, which `dot` & `vdot` don't have so i'd expect they're already generating device checks. those checks aren't enabled for all backends, though; according to [this comment](https://github.com/pytorch/pytorch/blob/50efbb9f1e7111b4b6d5b8e9a6064ee9783930be/torchgen/model.py#L1261-L1264), CUDA & HIP are the only in-tree backends that'll generate them automatically, which i guess is a divergence from the current behaviour of checking regardless of dispatched backend.

i could be wrong on this though, that's all just based on poking around a bit in the generated code.

Reply:
OK, I will add the device check back in